bin/run-command.sh

```diff
@@ -18,10 +18,71 @@ set -euo pipefail
 # TODO: Add ability to change task IAM Role. Part 3 of multipart update https://github.com/navapbc/template-infra/issues/354#issuecomment-1693973424
 # TODO: Change to keyword arguments. Part 3 of multipart update https://github.com/navapbc/template-infra/issues/354#issuecomment-1693973424
 
-APP_NAME="$1"
-ENVIRONMENT="$2"
-COMMAND="$3"
-ENVIRONMENT_VARIABLES=${4:-""}
+ACCOUNT_ID="$(./bin/current-account-id.sh)"
+
+while :; do
+  # echo "$1 $2"
+  if [ -z ${1+x} ]; then
+    break
+  fi
+  case $1 in
+    --environment-variables)
+      if [ "$2" ]; then
+        ENVIRONMENT_VARIABLES=$2
+        shift 2
+      else
+        echo "$1 flag present without Environment Variables"
+        exit 1
+      fi
+      ;;
+    --role-name)
+      if [ -z ${ROLE_ARN+x} ]; then
+        if [ "$2" ]; then
+          ROLE_ARN="arn:aws:iam::$ACCOUNT_ID:role/$2"
+          shift 2
+        else
+          echo "$1 flag present without role name"
+          exit 1
+        fi
+      else
+        printf "Conflicting attempts to define role\n%s\n%s" "$ROLE_ARN" "$1"
+        exit 1
+      fi
+      ;;
+    --role-arn)
+      if [ -z ${ROLE_ARN+x} ]; then
+        if [ "$2" ]; then
```

Comment:
this check, while well intentioned, doesn't actually check what it should be checking. if we add the option but forget to add a value and just move on to the next option, it doesn't catch the right error. see screenshot below for an example.

<img width="727" alt="image" src="https://github.com/navapbc/platform-test/assets/447859/b341267c-8963-4a5c-8f9d-53e5844e699c">

you may want to look into `get-opts` (see https://www.baeldung.com/linux/bash-parse-command-line-arguments)

Reply:
Good catch

Reply:
I looked into getopts, but it only handles single character arguments, and the GNU command get-opt seems a bit more complicated than it's worth. I would be open to just not do error handling for this